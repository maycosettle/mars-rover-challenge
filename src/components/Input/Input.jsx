import PropTypes from 'prop-types';

import useStyles from './styles';

const Input = ({ label, id, name, placeholder, onChange }) => {
  const styles = useStyles();

  return (
    <>
      <label htmlFor="label" className={styles.label}>
        {label}
      </label>
      <div className={styles.inputContainer}>
        <input type="text" name={name} id={id} className={styles.input} placeholder={placeholder} onChange={onChange} />
      </div>
    </>
  );
};

Input.propTypes = {
  label: PropTypes.string,
  id: PropTypes.string,
  name: PropTypes.string,
  placeholder: PropTypes.string,
  onChange: PropTypes.func,
};

Input.defaultProps = {
  label: '',
  id: '',
  name: '',
  placeholder: '',
  onChange: () => {},
};

export default Input;
