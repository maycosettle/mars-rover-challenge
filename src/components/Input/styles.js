const useStyles = () => ({
  label: 'block text-sm font-medium text-gray-700',
  inputContainer: 'mt-1',
  input: 'shadow-sm focus:ring-primary-500 focus:border-primary-500 block w-full sm:text-sm border-gray-300 rounded-md',
});

export default useStyles;
