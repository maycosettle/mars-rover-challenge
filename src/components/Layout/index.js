export * from './PageLayout';
export * from './PaperLayout';
export * from './PhotoLayout';
