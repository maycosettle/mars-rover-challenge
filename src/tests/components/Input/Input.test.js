import { shallow } from 'enzyme';
import { Input } from 'components/Input';

describe('tests on <Input />', () => {
  const wrapper = shallow(<Input />);

  test('Should render Input correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
