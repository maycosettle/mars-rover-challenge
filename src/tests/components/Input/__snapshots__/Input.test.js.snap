// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests on <Input /> Should render Input correctly 1`] = `
<Fragment>
  <label
    className="block text-sm font-medium text-gray-700"
    htmlFor="label"
  />
  <div
    className="mt-1"
  >
    <input
      className="shadow-sm focus:ring-primary-500 focus:border-primary-500 block w-full sm:text-sm border-gray-300 rounded-md"
      id=""
      name=""
      onChange={[Function]}
      placeholder=""
      type="text"
    />
  </div>
</Fragment>
`;
